kani-compiler/src/codegen_cprover_gotoc/codegen/assert.rs

```diff
@@ -346,41 +350,35 @@ impl<'tcx> GotocCtx<'tcx> {
             let ptr_place_ty = self.place_ty_stable(&ptr_place);
             if ptr_place_ty.kind().is_raw_ptr() {
                 // Extract the size of the pointee.
-                let pointee_size = {
-                    let TypeAndMut { ty: pointee_ty, .. } =
-                        ptr_place_ty.kind().builtin_deref(true).unwrap();
-                    let pointee_ty_layout = pointee_ty.layout().unwrap();
-                    pointee_ty_layout.shape().size.bytes()
-                };
+                let SizeAlign { size: sz, .. } =
```

Comment:
we should also check alignment.

Reply:
I checked, and it seems the alignment check is instrumented by the compiler when performing the dereference. For example, the following test:
```rust
#[repr(align(4))]
#[derive(Clone, Copy)]
struct AlignedI32(i32);

#[kani::proof]
fn f() {
    let data = AlignedI32(42);
    let ptr = &data as *const AlignedI32;
    
    let _ = unsafe { *ptr.byte_add(1) };
}
```
fails today with:
```

Check 3: f.assertion.1
         - Status: FAILURE
         - Description: "misaligned pointer dereference: address must be a multiple of its type's alignment"
         - Location: test.rs:10:22 in function f
```
Are there more alignment checks needed?

Reply:
But not the casting though, does it?

Reply:
Oh, I see. I'll look into checking alignment.

Reply:
Added in https://github.com/model-checking/kani/pull/3513/commits/47811273c88689757db0f5a03b80518430b0a6c8.